db/migrate/20200911083030_use_hash_index_for_reports.rb

```diff
@@ -0,0 +1,11 @@
+class UseHashIndexForReports < ActiveRecord::Migration[6.0]
+  def change
+    remove_index(:messages, :digest)
+    remove_index(:sources, :digest)
+    remove_column(:messages, :digest)
+    remove_column(:sources, :digest)
+
+    add_index(:messages, :value, using: 'hash')
+    add_index(:sources, :value, using: 'hash')
+  end
+end
```

Comment:
Adding index can be lengthy operation on big instances, we should probably implement a maintain check or something like that asking users to drop old reports prior upgrading.